packages/pane-view/src/PaneView.js

```diff
@@ -0,0 +1,283 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+
+import { buildClassName } from '@talixo/shared'
+
+import { roundDecimals, convertToPercent } from '../utils/utils'
+
+import Resizer from './Resizer'
+
+const moduleName = 'pane-view'
+
+const propTypes = {
+  /** Array of Pane components */
+  children: PropTypes.arrayOf(PropTypes.node).isRequired,
+
+  /** Additional class name */
+  className: PropTypes.string,
+
+  /** Function fired when Pane is resized */
+  onResize: PropTypes.func,
+
+  /** Function fired when Pane is started to drag */
+  onDragStart: PropTypes.func,
+
+  /** Function fired when Pane is dragged */
+  onDragResize: PropTypes.func,
+
+  /** Function fired when Pane is stopped to drag */
+  onDragStop: PropTypes.func,
+
+  /** Split direction */
+  split: PropTypes.oneOf(['horizontal', 'vertical']),
+
+  /** Additional styles for wrapper */
+  style: PropTypes.object
+}
+
+const defaultProps = {
+  split: 'horizontal'
+}
+
+/**
+ * Component which represents PaneView.
+ *
+ * @param {object} props
+ * @param {node} props.children
+ * @param {string} [props.className]
+ * @param {function} [props.onResize]
+ * @param {function} [props.onDragStart]
+ * @param {function} [props.onDragResize]
+ * @param {function} [props.onDragStop]
+ * @param {string} [props.split]
+ * @param {object} [props.style]
+ * @returns {React.Element}
+ */
+class PaneView extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      current: null,
+      paneArr: []
+    }
+
+    this.handleMouseDown = this.handleMouseDown.bind(this)
+    this.handleMouseMove = this.handleMouseMove.bind(this)
+    this.handleMouseUp = this.handleMouseUp.bind(this)
+    this.bindEventListeners = this.bindEventListeners.bind(this)
+    this.unbindEventListeners = this.unbindEventListeners.bind(this)
+  }
+
+  /**
+   * Set sizes for all childrens
+   */
+  componentDidMount () {
+    const { children } = this.props
+
+    if (children === undefined) return
+
+    let defaultSizeAll = 0
+    let numberOfNotDefaults = 0
+
+    children.forEach((el) => {
+      if (el.props.defaultSize !== undefined) {
+        defaultSizeAll += el.props.defaultSize
+      } else {
+        numberOfNotDefaults += 1
+      }
+    })
+
+    const defaultSizeSingle = roundDecimals((100 - defaultSizeAll) / numberOfNotDefaults)
+
+    const paneArr = children.map((el, i) => ({
+      paneId: i,
+      size: el.props.defaultSize !== undefined
+        ? el.props.defaultSize
+        : defaultSizeSingle
+    }))
+
+    this.setState({
+      paneArr
+    })
+  }
+
+  /**
+   * Handle componentWillUnmount
+   */
+  componentWillUnmount () {
+    this.unbindEventListeners()
+  }
+
+  /**
+   * Bind event listeners
+   */
+  bindEventListeners () {
```

Comment:
minor: I think it's more adding event listeners rather than binding (which are also present in this component)

```suggestion
  addEventListeners () {
```